doc/user/content/sql/types/map.md

```diff
@@ -0,0 +1,206 @@
+---
+title: "map Data Type"
+description: "Expresses a map"
+menu:
+  main:
+    parent: 'sql-types'
+aliases:
+  - /sql/types/map
+---
+
+{{< version-added v0.5.3 >}}
+
+`map` data expresses an unordered map with [`text`](../text) keys and an
+arbitrary uniform value type.
+
+Detail | Info
+-------|------
+**Quick Syntax** | `'{a=>123.4, b=>111.1}'::map[text=>double]'`
+**Size** | Variable
+
+## Syntax
+
+{{< diagram "type-map.svg" >}}
+
+Field | Use
+------|-----
+_map&lowbar;string_ | A well-formed map object.
+_value&lowbar;type_ | The [type](../../types) of the map's values.
+
+## Map operators
```

Comment:
This head feels redundant with the following `### Operators`. Looks like it's patterned after other types' "Function + Operator" sections? Until we have functions for it, let's elide the extra heading, and when we add the first function, we can add it in.